src/transactions/transferFeedUtils.ts

```diff
@@ -400,3 +426,56 @@ export function isTransferTransaction(
 ): tx is TransferItemFragment {
   return (tx as TransferItemFragment).address !== undefined
 }
+
+// Note: This hook is tested from src/transactions/feed/TransferFeedItem.test.ts
+function useFiatConnectTransferDisplayInfo({ amount, transactionHash }: TokenTransfer) {
+  const { t } = useTranslation()
+  const tokenInfo = useTokenInfo(amount.tokenAddress)
+  const fcTransferDetails = useSelector(getCachedFiatConnectTransferSelector(transactionHash))
+  const cachedFiatAccountUses = useSelector(cachedFiatAccountUsesSelector)
+  const account = useMemo(
```

Comment:
it appears that the "account" variable is serving two purposes now:
- storing information about the fiat account type that was used in a fiatconnect transfer
- acting as a flag for whether the transfer type is supported yet in the transaction feed, by being undefined if it was a transfer in, even if the user does actually have a fiat account id matching one in the fiatconnect transfer details

It makes the code more complex and hard to maintain to have one thing serve multiple purposes. I recommend finding another way to deal with transfers in.

Reply:
Ok cool, yeah I see how the changes you suggest make the code a bit cleaner/easier to reason about. Updated it